graylog2-web-interface/src/views/components/sidebar/create/AddWidgetButton.tsx

```diff
@@ -128,77 +121,110 @@ class AddWidgetButton extends React.Component<Props, State> {
 
       return () => {
         const id = generateId();
-        const onClose = () => this.setState((state) => {
-          const { overflowingComponents } = state;
 
-          delete overflowingComponents[id];
+        const onClose = () => {
+          setOverflowingComponents((cur) => {
+            const newState = { ...cur };
+
+            delete newState[id];
+
+            return newState;
+          });
 
           onClick();
+        };
 
-          return { overflowingComponents };
-        });
         const renderedComponent = <CreatorComponent key={creator.title} onClose={onClose} />;
 
-        this.setState((state) => {
-          const { overflowingComponents } = state;
+        setOverflowingComponents((cur) => {
+          const newState = { ...cur };
 
-          overflowingComponents[id] = renderedComponent;
+          newState[id] = renderedComponent;
 
-          return { overflowingComponents };
+          return newState;
         });
       };
     }
 
     throw new Error(`Invalid binding for creator: ${JSON.stringify(creator)} - has neither 'func' nor 'component'.`);
   };
 
-  _createMenuItem = (creator: Creator): React.ReactNode => {
-    const disabled = creator.condition?.() === false;
-
-    return (
-      <WithDispatch key={creator.title}>
-        {(dispatch) => (
-          <CreateButton key={creator.title}
-                        onClick={this._createHandlerFor(dispatch, creator)}
-                        disabled={disabled}>
-            {creator.title}
-          </CreateButton>
-        )}
-      </WithDispatch>
-    );
-  };
+  return (
+    <WithDispatch key={creator.title}>
+      {(dispatch) => (
+        <CreateButton key={creator.title}
+                      onClick={createHandlerFor(dispatch)}
+                      disabled={disabled}>
+          {creator.title}
+        </CreateButton>
+      )}
+    </WithDispatch>
```

Comment:
I tried to get rid of `WithDispatch`, but had problems using `dispatch` directly, because nothing happened when clicking on the button.